File: Letchu/TwoSum.java
import java.util.HashMap;
public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int i = 0;
        while (i < nums.length) {
            int x = target - nums[i];
            if (map.containsKey(x)) {
                return new int[] { i, map.get(x) };
            }
            map.put(nums[i], i);
            ++i;
        }
        return new int[] {};
    }
}

File: Jaggu/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
            
        for i in range(len(nums)):
            match_val = target - nums[i]
            if match_val in hashMap and i != hashMap[match_val]:
                return [hashMap[match_val], i]
            hashMap[nums[i]] = i
            
        

        